### ISSUE TITLE
Create a single config object for pages

### ISSUE THREAD

**Issue by thephilgray** [thephilgray/gulp-epub-starter#19]
With an app or website, it's possible to configure a build system to watch a directory of templates or views and build out routes based on the path of `index.html` files. 

This approach does not have a lot of use in the case of building EPUBs. Currently, gulp watches the `pages` directory while another task watches the output xhtml and sets the order of the pages in the `opf` file based on the default file organization (by name). This means that the filename of a template determines its order in the ebook, which creates some limitations and the need to use workarounds. 

One of those workarounds is to prefix the filename of every template. In one project, I simple named every template `page` + a number, as in `page01`. In another project, I simply prefixed every template with a 3 digit number, as in `001`, `002`, etc.

We are already using properties in the main config object to manually set things like page properties (whether a page is scripted or not), as well as which pages should be included in the TOC. My first thought was to store all this information in something like YAML properties at the top of each template. There are ways of using YAML with pug. However, since the the script that compiles the `opf` file runs after the pug is compiled to xhtml, I haven't found an easy way to extract these values. There could be an argument for using some enhanced version of Markdown as the templating language. But in the meantime, it would make sense to maintain one object with all the configurable page data. 

The downside is that it will be add complexity and be less intuitive. But the advantages are a lot in terms of flexibility. With the pages config object, it would be a lot easier to establish hierarchical relationships (like chapter) which could be used to automate more advanced TOCs, set special page properties, make certain files optional depending on the device/layout (like cover.xhtml, for instance).

Ideally, it makes sense to marry the template with the template config/props as done successfully by numerous frontend frameworks and libraries. But in the meantime, separate configuration seems like the easiest path to having a tool that can handle different projects.

**Comment by thephilgray** [comment]
One possible route is to create a scaffolding tool that builds the initial project based on a json (or csv) file. This way, you could start with templates in place. It should copy the same json file into the scaffolded project so you can edit it at any time, injecting different data/properties into your templates. 

Another command can add pages/templates to an already existing project. This would add entries to the json file on your behalf and scaffold out the new templates. It would be nice to be able to run this type of command without restarting the dev script, but some other things need to happen before that's possible.

**Comment by thephilgray** [comment]
For user-friendliness, it could use YAML. As in: 

```yaml
meta:
  title: Lorem Title
  cover: ../images/cover.jpg
  date: 2018-08-22T01:47:08-04:00
  author: Lorem Author
  identifier: urn:uuidRV1052018
  language: en-US
  publisher: Lorem Publisher
pages:
  - '001':
      chapter: Chapter 1
      title: Page One
      toc: true
      properties:
        - scripted
```

The build process could first look for this file, and if none exists, then generate one by scraping values from the src files ?
